metaflow/plugins/pypi/parsers.py

```diff
@@ -0,0 +1,260 @@
+    python_version = parse_req_value(
+        config_value, extra_args, sources, deps, np_deps, sys_deps
+    )
+    result = {}
+    if python_version:
+        result["python"] = python_version
+
+    if extra_args:
+        raise ValueError(
+            "Additional arguments are not supported when parsing requirements.txt for "
+            "the pypi decorator -- use Netflix's Metaflow extensions (metaflow-netflixext)'s "
+            "named environment instead"
+        )
+    if np_deps:
+        raise ValueError(
+            "Non-python dependencies are not supported when parsing requirements.txt for "
+            "the pypi decorator -- use Netflix's Metaflow extensions (metaflow-netflixext)'s "
+            "named environment instead"
+        )
+    if sys_deps:
+        raise ValueError(
+            "System dependencies are not supported when parsing requirements.txt for "
+            "the pypi decorator -- use Netflix's Metaflow extensions (metaflow-netflixext)'s "
+            "named environment instead"
+        )
+
+    if sources:
+        raise ValueError(
+            "Specifying extra indices is not supported. Include those sources "
+            "directly in your pip.conf or use Netflix's Metaflow extensions "
+            "(metaflow-netflixext)"
+        )
+
+    result["packages"] = deps
+
+    return result
+
+
+def yml_parser(config_value: str) -> Dict[str, Any]:
+    sources = {}
+    conda_deps = {}
+    pypi_deps = {}
+    sys_deps = {}
+    python_version = parse_yml_value(
+        config_value, {}, sources, conda_deps, pypi_deps, sys_deps
+    )
+    result = {}
+    if sys_deps:
+        raise ValueError(
+            "System dependencies are not supported when parsing environment.yml for "
+            "the conda decorator -- use Netflix's Metaflow extensions (metaflow-netflixext)'s "
+            "named environment instead"
+        )
+
+    if python_version:
+        result["python"] = python_version
+
+    if sources:
+        raise ValueError(
+            "Channels or extra indices are not supported when parsing environment.yml for "
+            "the conda decorator -- use Netflix's Metaflow extensions (metaflow-netflixext) "
+            "or specify CONDA_CHANNELS (for channels) and set indices in your pip.conf"
+        )
+    if pypi_deps:
+        raise ValueError(
+            "Mixing conda and pypi packages is not supported when parsing environment.yml for "
+            "the conda decorator -- use Netflix's Metaflow extensions (metaflow-netflixext) "
+            "or stick to only one ecosystem"
+        )
+
+    if len(conda_deps):
+        result["libraries"] = conda_deps
+
+    return result
+
+
+def parse_req_value(
+    file_content: str,
+    extra_args: Dict[str, List[str]],
+    sources: Dict[str, List[str]],
+    deps: Dict[str, str],
+    np_deps: Dict[str, str],
+    sys_deps: Dict[str, str],
+) -> Optional[str]:
+    from packaging.requirements import InvalidRequirement, Requirement
```

Comment:
only change to the original here, as `packaging` is not available by default in environments, having the import trickle down to the top-level causes issues.